src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,359 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NLog;
+using FluentValidation.Results;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.RemotePathMappings;
+using Tribler.Api;
+using System.Net.Http.Headers;
+using System.Collections.Generic;
+using System.Linq;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Validation;
+using MonoTorrent;
+using System.Net;
+using NzbDrone.Common.Serializer;
+
+namespace NzbDrone.Core.Download.Clients.Tribler
+{
+    using TriblerApi = global::Tribler.Api;
+
+    public class TriblerDownloadClient : TorrentClientBase<TriblerDownloadSettings>
+    {
+        public TriblerDownloadClient(ITorrentFileInfoReader torrentFileInfoReader,
+                            IHttpClient httpClient,
+                            IConfigService configService,
+                            IDiskProvider diskProvider,
+                            IRemotePathMappingService remotePathMappingService,
+                            Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+        }
+
+        public override string Name => "Tribler";
+
+        public override bool PreferTorrentFile => false;
+
+        public string GetBaseUrl(string relativePath = null)
+        {
+            var baseUrl = HttpRequestBuilder.BuildBaseUrl(Settings.UseSsl, Settings.Host, Settings.Port, Settings.UrlBase);
+            baseUrl = HttpUri.CombinePath(baseUrl, relativePath);
+
+            return baseUrl;
+        }
+
+        private HttpRequestBuilder getRequestBuilder(string relativePath = null)
```

Comment:
Request functions should be part of a separate Proxy class, separating client communication from processing those items and the business logic, see `UTorrent` and `UTorrentProxy` as an example.